### PR TITLE
Update package build to support ROS Noetic

### PR DESCRIPTION
**Pull request by ahcorde**
Following the guidelines to migrate packages to [noetic](http://wiki.ros.org/noetic/Migration)

- import setup from setuptools instead of distutils-core

- Bump CMake version to avoid CMP0048 warning
This bumps the minimum CMake version to 3.0.2, which is the [minimum supported by ROS Kinetic](https://www.ros.org/reps/rep-0003.html#kinetic-kame-may-2016-may-2021) and new enough to default to the NEW behavior of [CMP0048](https://cmake.org/cmake/help/v3.0/policy/CMP0048.html). This avoids a CMake warning when building and testing this package in Debian Buster and Ubuntu Focal.

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
+try:
+    from distutils.core import setup
+except ImportError:
+    from setuptools import setup
```
Comment (sloretz) [inline setup.py:6]: I think this should just import setuptools with conditional dependencies

https://github.com/ros-controls/ros_control/pull/429/files#diff-f3ba5e3bc4f4fc2a8ce3f52bfc533377

Comment (cottsay) [reply setup.py:6]: If this is OK as far back as Kinetic, then :+1:. It would also align better with the ros2 branches.

Comment (cottsay) [reply setup.py:6]: There is only one package under `desktop_full` in Kinetic that unconditionally imports setuptools. If it's all the same, I'd prefer to maintain compatibility with distutils.

That said, we should invert the logic and prefer setuptools, as was done here: https://github.com/ros-visualization/python_qt_binding/commit/0a2e678b141c2802b0b83b562bfd8d075a0535f7

File: package.xml [hunk 2]
```diff
@@ -25,6 +28,8 @@
   <author>Ze'ev Klapow</author>
 
   <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend condition="$ROS_PYTHON_VERSION == 2">python-setuptools</buildtool_depend>
+  <buildtool_depend condition="$ROS_PYTHON_VERSION == 3">python3-setuptools</buildtool_depend>
```
Comment (cottsay) [inline package.xml:32]: There are no other packages declaring a dependency on `setuptools` in the `package.xml`. Plus, this makes the dependency required, which makes the fallback to distutils in the `setup.py` pointless. I think this change should be dropped.

**Comment by cottsay** [comment]
> Looks like this package still depends on a Python 2 dependency. It should depend on python3-yaml in Noetic instead.

I opened #78 to address this.

File: package.xml (hunk 2 above)
Comment (cottsay) [reply package.xml:32]: Ah, I was mistaken, this is only true of Kinetic. Looks like this is common in Noetic.

Comment (ahcorde) [reply package.xml]: done

Comment (cottsay) [reply package.xml]: Sorry! I think the change is OK! I just jumped the gun. Bring it back! Bring it back!

File: setup.py (hunk 1 above)
Comment (cottsay) [reply setup.py:4]: Now that I've had my coffee and the conditions were added, I'm on board with @sloretz. At least one other package is calling for setuptools directly in Kinetic, so I think we can do the same.

Sorry for the back-and-forth - maintaining a single branch for all of the releases can be tricky.

Last thing, and then we can merge this!

**Pull request merged**